### PR TITLE
gk: change scan interval unit to iterations

### PR DESCRIPTION
**Pull request by cjdoucette**
If you want me to run performance tests before we merge Qiaobin's changes, just let me know. Otherwise, I will wait until Qiaobin's PR is merged and then I will rebase and test these changes.

### CODE REVIEW

**Comment by AltraMayor** [comment]
I just merged Qiaobin's pull request, so you can go ahead with the rebase and evaluation.

File: gk/main.c [hunk 1]
```diff
@@ -1971,31 +1971,19 @@ gk_proc(void *arg)
...
-		(double)(gk_conf->flow_table_full_scan_ms *
-		rte_get_tsc_hz()) / (num_buckets * 1000.));
-	if (bucket_scan_timeout_cycles == 0) {
-		GK_LOG(WARNING,
-			"The value of the field flow_table_full_scan_ms in Gatekeeper configuration is too small\n");
-		exiting = true;
-		return -1;
-	}
+	uint32_t scan_iter = gk_conf->flow_table_scan_iter;
+	uint64_t iter_count = 0;
```
Comment (AltraMayor) [inline gk/main.c]: `iter_count` is never bigger than `scan_iter`, so `iter_count` can be of type `uint32_t`.

**Comment by cjdoucette** [comment]
Before this patch, the performance is:

| lcores | table size | GK Mpps rcvd | GK Gibps rcvd | client Mpps sent | client Gibps sent |
|--------|------------|--------------|---------------|------------------|-------------------|
|  1  |  2^10  |  3.93  |  1.87  |  6.05  |  2.89  |
|  1  |  2^15  |  4.0  |  1.91  |  6.80  |  3.24  |
|  1  |  2^20  |  3.2  |  1.53  |  5.94  |  2.83  |
|        |            |              |              |                  |                  |
|  2  |  2^10  |  5.88  |  2.8  |  5.94  |  2.83  |
|  2  |  2^15  |  5.99  |  2.86  |  6.16  |  2.94  |
|  2  |  2^20  |  5.44  |  2.59  |  5.48  |  2.61  |
|        |            |              |              |                  |                  |
|  3  |  2^10  |  5.89  |  2.81  |  5.78  |  2.76  |
|  3  |  2^15  |  5.79  |  2.76  |  5.91  |  2.82  |
|  3  |  2^20  |  5.47  |  2.61  |  5.95  |  2.84  |
|        |            |              |              |                  |                  |

After this patch, the performance is:

| lcores | table size | GK Mpps rcvd | GK Gibps rcvd | client Mpps sent | client Gibps sent |
|--------|------------|--------------|---------------|------------------|-------------------|
|  1  |  2^10  |  3.75  |  1.79  |  6.69  |  3.19  |
|  1  |  2^15  |  3.76  |  1.79  |  6.47  |  3.08  |
|  1  |  2^20  |  3.79  |  1.8  |  6.22  |  2.96  |
|        |            |              |              |                  |                  |
|  2  |  2^10  |  6.06  |  2.89  |  6.12  |  2.92  |
|  2  |  2^15  |  6.12  |  2.92  |  6.11  |  2.91  |
|  2  |  2^20  |  5.84  |  2.78  |  5.88  |  2.80  |
|        |            |              |              |                  |                  |
|  3  |  2^10  |  6.35  |  3.03  |  6.37  |  3.04  |
|  3  |  2^15  |  6.37  |  3.04  |  6.43  |  3.06  |
|  3  |  2^20  |  5.48  |  2.61  |  5.93  |  2.83  |
|        |            |              |              |                  |                  |

**Comment by cjdoucette** [comment]
The pull request is ready for another review.

**Pull request merged**